filebeat/scripts/generator/generator.go

```diff
@@ -70,7 +70,8 @@ func copyTemplate(template, dest string, replace map[string]string) error {
 	if err != nil {
 		return fmt.Errorf("cannot copy template: %v", err)
 	}
-	return nil
+
+	return os.Chmod(dest, 0644)
```

Comment:
`0644` should be passed to the `ioutil.WriteFile` rather than doing this in a separate syscall.

On a side note, it looks like this file is is incorrectly using `os.ModePerm`. Generally this value is used as a mask with mode bits to isolate the permission bits (e.g. `perm = fileinfo.Mode() & os.ModePerm`). Writing a file with `0777` should almost never be done for security reasons. In most cases if you are creating a file without any sensitive data use `0644` and if it's a dir use `0755`.